modServer.bas

```diff
@@ -3481,85 +3481,6 @@ Function NoDirectionalWalls(TheMap As Long, X As Long, Y As Long, Direction As L
     End Select
 End Function
 
-Function NoDirectionalWalls2(TheMap As Long, X As Long, Y As Long, Direction As Long) As Boolean
-    If X < 0 Or Y < 0 Or X > 11 Or Y > 11 Then
-        NoDirectionalWalls2 = False
-        Exit Function
-    End If
-
-    NoDirectionalWalls2 = True
-    Select Case Direction
-    Case 1    'Up
-        If Y >= 0 Then
-            If Y > 0 Then
-                If Map(TheMap).Tile(X, Y - 1).Att = 17 Then
-                    If ExamineBit(Map(TheMap).Tile(X, Y - 1).AttData(0), 3) Then
-                        NoDirectionalWalls2 = False
-                        Exit Function
-                    End If
-                End If
-            End If
-            If Map(TheMap).Tile(X, Y).Att = 17 Then
-                If ExamineBit(Map(TheMap).Tile(X, Y).AttData(0), 1) Then
-                    NoDirectionalWalls2 = False
-                    Exit Function
-                End If
-            End If
-        End If
-    Case 0    'Down
-        If Y < 12 Then
-            If Y < 11 Then
-                If Map(TheMap).Tile(X, Y + 1).Att = 17 Then
-                    If ExamineBit(Map(TheMap).Tile(X, Y + 1).AttData(0), 0) Then
-                        NoDirectionalWalls2 = False
-                        Exit Function
-                    End If
-                End If
-            End If
-            If Map(TheMap).Tile(X, Y).Att = 17 Then
-                If ExamineBit(Map(TheMap).Tile(X, Y).AttData(0), 2) Then
-                    NoDirectionalWalls2 = False
-                    Exit Function
-                End If
-            End If
-        End If
-    Case 3    'Left
-        If X >= 0 Then
-            If X > 0 Then
-                If Map(TheMap).Tile(X - 1, Y).Att = 17 Then
-                    If ExamineBit(Map(TheMap).Tile(X - 1, Y).AttData(0), 6) Then
-                        NoDirectionalWalls2 = False
-                        Exit Function
-                    End If
-                End If
-            End If
-            If Map(TheMap).Tile(X, Y).Att = 17 Then
-                If ExamineBit(Map(TheMap).Tile(X, Y).AttData(0), 4) Then
-                    NoDirectionalWalls2 = False
-                    Exit Function
-                End If
-            End If
-        End If
-    Case 2    'Right
-        If X < 12 Then
-            If X < 11 Then
-                If Map(TheMap).Tile(X + 1, Y).Att = 17 Then
-                    If ExamineBit(Map(TheMap).Tile(X + 1, Y).AttData(0), 5) Then
-                        NoDirectionalWalls2 = False
-                        Exit Function
-                    End If
-                End If
-            End If
-            If Map(TheMap).Tile(X, Y).Att = 17 Then
-                If ExamineBit(Map(TheMap).Tile(X, Y).AttData(0), 7) Then
-                    NoDirectionalWalls2 = False
-                    Exit Function
-                End If
-            End If
-        End If
-    End Select
-End Function
-

```

Comment:
I just checked [Remote's original source](https://github.com/TheRemote/OdysseyClassic/blob/master/Server/modServer.bas) and these lines are included in there. So it's just random buggy code with no apparent use? How strange.

Reply:
Yeah, I dunno who put them there but using the NoDirectionalWalls2 function is currently giving an in-game error for ungodded players, causing them to warp back rather than walk over when attempting to walk up or down on a directional wall. This error was raised by player in-game name: Robz, inside his house.

Reply:
Yeah I've noticed it for awhile, I couldn't figure out what was causing it. Only for ungodded players, so strange.

Reply:
The server doesn’t check for walls for godded players.